nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -826,8 +833,11 @@ def fwd_output_and_loss_func(dataloader_iter, model, checkpoint_activations_all_
                 'labels': batch['labels'],
                 'loss_mask': batch['loss_mask'],
             }
+
             if not self.mcore_gpt:
                 forward_args['checkpoint_activations_all_layers'] = checkpoint_activations_all_layers
+                if not self.use_loss_mask:
+                    forward_args.pop('loss_mask')
```

Comment:
@ericharper  do we need yet another flag here? why not just pop loss_mask until mcore supports it?